tests/unit/remote/ssh/test_pool.py

```diff
@@ -18,3 +26,87 @@ class MyError(Exception):
             key_filename=TEST_SSH_KEY_PATH,
         ):
             raise MyError
+
+
+@pytest.mark.parametrize(
+    "uname,md5command", [("Linux", "md5sum"), ("Darwin", "md5")]
+)
+def test_escapes_filepaths_for_md5_calculation(
+    ssh_server, uname, md5command, mocker
+):
+    fake_md5 = "x" * 32
+
+    with get_connection(
+        SSHConnection,
+        host=ssh_server.host,
+        port=ssh_server.port,
+        username=TEST_SSH_USER,
+        key_filename=TEST_SSH_KEY_PATH,
+    ) as connection:
+        uname_mock = mocker.PropertyMock(return_value=uname)
+        mocker.patch.object(SSHConnection, "uname", new_callable=uname_mock)
+        connection.execute = mocker.Mock(return_value=fake_md5)
+        connection.md5(SRC_PATH_WITH_SPECIAL_CHARACTERS)
+        connection.execute.assert_called_with(
+            f"{md5command} {ESCAPED_SRC_PATH_WITH_SPECIAL_CHARACTERS}"
+        )
```

Comment:
Seems like for all of these tests we don't really need to create a real connection, we could've just mocked `__init__` and `self.execute`, and that should be enough. Just to make the tests a bit lighter :slightly_smiling_face: What do you think?

Reply:
Also, let's move them to `tests/unit/remote/ssh/test_connection.py`

Reply:
Looks like the tests in `test_connection.py` just have the ssh_connection injected anyway, so how about I just do the same? (See latest commit)

Reply:
That will do :slightly_smiling_face: Thanks!